.github/workflows/query-list.yml

```diff
@@ -0,0 +1,39 @@
+name: Build code scanning query list
+
+on:
+  push:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Clone self (github/codeql)
+      uses: actions/checkout@v2
+      with:
+        path: codeql 
+    - name: Clone github/codeql-go
+      run: git clone https://github.com/github/codeql-go
```

Comment:
You can use `actions/checkout` again here, just specify `path: codeql-go`. The advantage is that it reduces the fetch depth.